content/blog/best_ngrok_alternatives.md

```diff
@@ -334,6 +345,28 @@ The **Tailscale Funnel** represents a functionality enabling the directing of ex
 
 Using Tailscale Funnel you can achive the functionality of Ngrok.
 
+#### Pros of Tailscale
+
+- Zero-config mesh VPN with WireGuard protocol
+- Point-to-point connectivity with no single point of failure
+- Excellent NAT traversal capabilities
+- Works across multiple platforms and cloud providers
+- Free for personal use (up to 100 devices)
+- Built-in MagicDNS for easy device naming
+- ACL-based access control
+
+#### Cons of Tailscale
+
+- Not a direct tunneling solution like Ngrok
+- Requires installation on all devices that need to connect
+- More complex setup for simple use cases
+- Tailscale Funnel feature is less mature than dedicated tunneling tools
+- Limited to VPN-style connectivity rather than public URL exposure
```

Comment:
Funnel can listen on limited public ports 443, 8443, and 10000.

no tcp or udp tunnels.